jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -810,12 +811,15 @@ public Set<String> getResourcePaths(String path)
             Resource resource = getResource(path);
 
             if (!path.endsWith("/"))
-                path = path + "/";
+                path = path + '/';
 
             HashSet<String> set = new HashSet<>();
             for (Resource item: resource.list())
             {
-                set.add(path + item.getFileName());
+                String entry = path + item.getFileName();
+                if (item.isDirectory())
+                    entry = entry + '/';
+                set.add(entry);
```

Comment:
Is it possible to resolve this within the resource classes?   Previously the resource classes would have the trailing '/' on directories so it would be everywhere.  Doing it just here as a post process risks not covering all usages.

Reply:
The Resource object returns a `List<Resource>` so I think perhaps the place would be `.getFileName()` ?

Reply:
Yes getFileName, toString and anything else that returns the path as a string. Ideally even the URI should have the trailing slash. But
I don't think Path retains it.

Reply:
Turns out fixing `getFileName()` breaks a lot of test cases. 
See latest Jenkins build on this PR.
I'll noodle through them on monday.

Reply:
Would be good to check if any of these texts existed in jetty 11 and see if they were changed to not expect the trailing slash

Reply:
The Jetty 11 `Resource.getName()` does not return the trailing slash on directories.

However, the Jetty 11 `Resource.list()` returns a String array, and that array has the trailing slash.
The `Resource.list()` method in Jetty 11 is used ...

* ContextHandler.getResourcePaths(String) which returns a `Set<String>` on the Servlet API
* Resource.getListHTML - used to iterate a directory, and build a new set of Resources, that eventually are presented (in HTML)
* MetaInfConfiguration - iterates over WEB-INF/lib/*.jar files, and builds a new set of Resources that eventually wind up in the WebAppClassLoader
* WebAppClassLoader - used in addJars(Resource) to iterate a directory and build a new set of Resources that are added to the WebAppClassLoader
* Runner.Classpath.addJars(Resource) - same a WebAppClassLoader.addJars(Resource)

Turns out that in Jetty 12 we skip the String array step (eg: the _Resource -> String array -> List of Resource -> iterate/process_ flow).
In Jetty 11, we don't rely on the results of `Resource.getName()` for anything critical, it seems to be used only for things like debugging, exception messages, keys in the annotation parser, etc.